src/arch/x86: Ensure $(objgenerated) exists before it's used

In some rare cases it seems that make tries to build
$(objgenerated)/assembly.inc before the build-dirs target has finished,
and so assembly.inc can't be written. Enforce that build-dirs is done
before assembly.inc starts.

BUG=chromium:1098215
BRANCH=none
TEST=none

Change-Id: Ib141ea45a43836cfdde0059523c331fe5286b06d
Signed-off-by: Patrick Georgi <pgeorgi@google.com>
Reviewed-on: https://review.coreboot.org/c/coreboot/+/42883
Tested-by: build bot (Jenkins) <no-reply@coreboot.org>
Reviewed-by: Aaron Durbin <adurbin@chromium.org>

diff --git a/src/arch/x86/Makefile.inc b/src/arch/x86/Makefile.inc
index 6297384..1a1aa40 100644
--- a/src/arch/x86/Makefile.inc
+++ b/src/arch/x86/Makefile.inc

@@ -50,7 +50,7 @@
 # into a single generated file.
 crt0s = $(cpu_incs-y)
 
-$(objgenerated)/assembly.inc: $$(crt0s)
+$(objgenerated)/assembly.inc: build-dirs $$(crt0s)
 	@printf "    GEN        $(subst $(obj)/,,$(@))\n"
 	printf '$(foreach crt0,$(crt0s),#include "$(crt0)"\n)' > $@
